bff/templates/deployment-sidecar.yaml

```diff
@@ -0,0 +1,104 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "bff.fullname" . }}-sidecar
+  {{- with .Values.deploymentAnnotationsSidecar }}
```

Comment:
The sidecar related properties should be nested in a `.Values.sidecar` group IMO.

For example:
`.Values.sidecar.deploymentAnnotation`

edit: same for everything else in this yaml file